test/final.test.js

```diff
@@ -147,3 +138,22 @@ test('also instruments custom log methods', async ({pass, fail, error}) => {
   await sleep(10)
   if (passed === false) fail('flushSync not called')
 })
+
+function testLoggers (logger, finalLogger, is, isNot) {
```

Comment:
I would much prefer that the assertions are repeated inline

for testing repetition > indirection - keep things declarative